packages/logging-bunyan/src/index.js

```diff
@@ -0,0 +1,134 @@
+/*!
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module logging-bunyan
+ */
+
+'use strict';
+
+var logging = require('@google-cloud/logging');
+
+/**
+ * Map of Stackdriver logging levels.
+ *
+ * @type {object}
+ * @private
+ */
+var BUNYAN_TO_STACKDRIVER = {
+  60: 'critical',
+  50: 'error',
+  40: 'warning',
+  30: 'info',
+  20: 'debug',
+  10: 'debug'
+};
+
+/**
+ * This module provides support for streaming your bunyan logs to
+ * [Stackdriver Logging]{@link https://cloud.google.com/logging}.
+ *
+ * If your app is running on Google Cloud Platform, all configuration and
+ * authentication is handled for you. We also auto-detect the appropriate
+ * resource descriptor to report the log entries against.
+ *
+ * If you are running your application in anther environment, such as locally,
+ * on-premise, or on another cloud provider, you will need to provide additional
+ * configuration.
+ *
+ * @constructor
+ * @alias module:logging-bunyan
+ *
+ * @param {object} options - [Configuration object](#/docs). Refer to this link
+ *     for authentication information.
+ * @param {string=} options.logName - The name of the log that will receive
+ *     messages written to this bunyan stream. Default: `bunyan_Log`.
+ * @param {object=} options.resource - The monitored resource that the log
+ *     stream corresponds to. On Google Cloud Platform, this is detected
+ *     automatically, but you may optionally specify a specific monitored
+ *     resource. For more infromation see the
+ *     [official documentation]{@link
+ * https://cloud.google.com/logging/docs/api/reference/rest/v2/MonitoredResource}
+ *
+ * @example
+ * var bunyan = require('bunyan');
+ * var loggingBunyan = require('@google-cloud/logging-bunyan')({
+ *    projectId: 'grape-spaceship-123',
+ *    keyFilename: '/path/to/keyfile.json',
+ *    resource: {
+ *       type: 'global'
+ *    }
+ * });
+ *
+ * var logger = bunyan.createLogger({
+ *   name: 'my-service',
+ *   streams: [loggingBunyan.stream('info')]
+ * });
+ *
+ */
+function LoggingBunyan(options) {
+  if (!(this instanceof LoggingBunyan)) {
+    return new LoggingBunyan(options);
+  }
+
+  options = options || {};
+
+  this.logName_ = options.logName || 'bunyan_log';
```

Comment:
Do we want the default name to make it clear that our module is generating the logs.